apps/chat/src/store/conversations/conversations.epics.ts

```diff
@@ -1113,76 +1114,109 @@ const sendMessageEpic: AppEpic = (action$, state$) =>
+        };
 
-      const assistantMessage: Message = {
-        content: '',
-        model: messageModel,
-        settings: messageSettings,
-        role: Role.Assistant,
-      };
+        let currentMessages =
+          payload.deleteCount > 0
+            ? payload.conversation.messages.slice(
+                0,
+                payload.deleteCount * -1 || undefined,
+              )
+            : payload.conversation.messages;
 
-      const userMessage: Message = {
-        ...payload.message,
-        model: messageModel,
-        settings: messageSettings,
-      };
+        const assistantMessage: Message = {
+          content: '',
+          model: messageModel,
+          settings: messageSettings,
+          role: Role.Assistant,
+        };
 
-      const updatedMessages: Message[] = (
-        payload.deleteCount > 0
-          ? payload.conversation.messages.slice(
-              0,
-              payload.deleteCount * -1 || undefined,
-            )
-          : payload.conversation.messages
-      ).concat(userMessage, assistantMessage);
+        const userMessage: Message = {
+          ...payload.message,
+          model: messageModel,
+          settings: messageSettings,
+        };
 
-      const newConversationName = getNextDefaultName(
-        getNewConversationName(
-          payload.conversation,
-          payload.message,
-          updatedMessages,
-        ),
-        conversations.filter(
-          (conv) =>
-            conv.folderId === payload.conversation.folderId &&
-            conv.id !== payload.conversation.id,
-        ),
-        Math.max(selectedConversationIds.indexOf(payload.conversation.id), 0),
-        true,
-      );
+        /*
+          Overlay needs to share host application state information
+          We storing state information in systemPrompt (message with role: Role.System)
+        */
+        if (overlaySystemPrompt) {
+          const overlaySystemPromptMessage: Message = {
+            content: overlaySystemPrompt,
+            role: Role.System,
+          };
 
-      const updatedConversation: Conversation = regenerateConversationId({
-        ...payload.conversation,
-        lastActivityDate: Date.now(),
-        replay: payload.conversation.replay
-          ? {
-              ...payload.conversation.replay,
-              activeReplayIndex: payload.activeReplayIndex,
-            }
-          : undefined,
-        messages: updatedMessages,
-        name: newConversationName,
-        isMessageStreaming: true,
-      });
+          // removing previous system message
+          currentMessages = currentMessages.filter(
+            (message) => message.role !== Role.System,
+          );
 
-      return {
-        oldConversationId: payload.conversation.id,
-        updatedConversation,
-        modelsMap,
-        assistantMessage,
-      };
-    }),
+          currentMessages.push(overlaySystemPromptMessage);
```

Comment:
Maybe it's better to replace previous system message and place new on same index?
Or if it was not presented place on zero place in this array?

Reply:
Fixed